ml_tools/tests/test_feature_generator.py

```diff
@@ -0,0 +1,201 @@
+import pytest
+from uszipcode import SearchEngine
+
+from ml_tools import FeatureGenerator, \
+    Hour, \
+    SimpleAggregator, \
+    custom_generator, \
+    SingleAggregator, \
+    Average, \
+    DateTimeInfo, \
+    ZipCodeInfo
+import pandas as pd
+
+
+def test_not_implemented():
+    with pytest.raises(TypeError, match="Can't instantiate abstract class FeatureGenerator with abstract methods "
+                                        "generate_feature"):
+        FeatureGenerator()
+
+
+# def test_feature_generator_init(single_dataset):
+#     df = single_dataset
+#
+#     test_feature = GenericGenerator(df)
+#
+#     assert test_feature._data.equals(df)
+
+
+def test_hour(timestamps):
+    df = timestamps
+    df_comp = df
+
+    for idx, time in enumerate(df.timestamp):
+        df_comp.timestamp[idx] = time.hour
+
+    new_feature = Hour.generate_feature(df)
+
+    assert new_feature.equals(df_comp)
+
+
+def test_hour_columns(timestamps):
+    df = timestamps
+    df_comp = df.copy()
+
+    for idx, time in enumerate(df.timestamp):
+        df_comp.timestamp[idx] = time.hour
+
+    new_feature = Hour.generate_feature(df, column='timestamp')
+
+    assert new_feature.equals(df_comp)
+
+
+def test_split_timestamp(timestamps):
+    df = timestamps
+    split_time = DateTimeInfo.generate_feature(df.copy(), 'timestamp')
+
+    assert all([d1.year == d2 for d1, d2 in zip(df['timestamp'], split_time['timestamp_year'])])
+    assert all([d1.month == d2 for d1, d2 in zip(df['timestamp'], split_time['timestamp_month'])])
+    assert all([d1.day == d2 for d1, d2 in zip(df['timestamp'], split_time['timestamp_day'])])
+    assert all([d1.weekday() == d2 for d1, d2 in zip(df['timestamp'], split_time['timestamp_weekday'])])
+    assert all([d1 == d2 for d1, d2 in zip(df['timestamp'].dt.time, split_time['timestamp_time'])])
+
+
+def test_zipcode_info(zipcodes):
```

Comment:
If you were keeping that exception your generators were raising, you'd want to test those too.  If you wanted to go another step, you could run your pytest file with code coverage to make sure you didn't miss any lines of code in your tests.